### DISCUSSION TITLE
Learning: Licensing for Open Science

### DISCUSSION THREAD

**Discussion by labarba** [gw-ospo/community#1] [Ideas]
**What**
We want to develop a short course that introduces researchers (doctoral students, postdocs, staff researchers and PIs) to the essentials of OSS licensing for research and educational purposes. The course duration should be about 3 hours when taught live, and we will also produce an online version for self-study. The syllabus should cover key topics like license types, compatibility, institutional processes, and considerations for researchers.

**Idea**
Organizes the topics in three sections that focus on: 
1. Use openly licensed works published by others: what you need to know. 
2. License your own works: what works for code, data, and other content types.
3. Collaborating with others in open source: license considerations for research teams.

Based on the suggested approach, here is a detailed outline for the short course organized into three main sections.

---

# Title: Open Source Software Licensing for Research and Education

## 1. Using Openly Licensed Works Published by Others (1 hour)
1. Introduction to open licensing concepts
        - Definition of open source, open data, open content
        - Importance of open licensing for research and education
2. Understanding license types
        - Proprietary licenses
        - Open source (permissive) licenses: MIT, BSD, Apache, etc.
        - Open source (copyleft) licenses: GPL, LGPL, MPL
3. License compatibility and composability
        - Understanding license compatibility
        - Implications of license directionality
        - Combining code, data, and content under different licenses
4. Practical considerations for using third-party open works
        - Identifying and complying with license terms
        - Attribution and citation requirements
        - Source code availability and modifiability

## 2. Licensing Your Own Works (1 hour)
1. Why license your research outputs?
        - Reproducibility, transparency, and scientific norms
        - Enabling collaboration and building on existing work
2. Licensing code, data, and other content types
        - Considerations for licensing software
        - Licensing options for research data
        - Creative Commons licenses for text, media, and other content
3. Choosing an appropriate open license
        - Factors to consider (use case, audience, commercialization, etc.)
        - Academic and research use cases
        - Institutional policies and processes
4. Working with the technology transfer office
        - Understanding institutional IP policies
        - Navigating the licensing process
        - Negotiating open licensing for research outputs

## 3. Collaborating in Open Source Research Teams (1 hour)
~1. Benefits and challenges of open source collaboration~
        - ~Community contributions and code sustainability~
        - ~Coordination and communication overhead~
~2. License considerations for collaborative projects
        - ~Inbound: understanding licenses of third-party components~
        - ~Outbound: choosing a license for collaborative outputs~
        - ~License proliferation and managing dependencies~
~3. Tools and practices for open source collaboration~
        - ~Version control systems (Git, GitHub, etc.)~
        - ~Contribution guidelines and governance models~
        - ~Open science frameworks and platforms~
~4. Building and sustaining an open source community~
        - ~Attracting and retaining contributors~
        - ~Leveraging open source for research dissemination and impact~
        - ~Open source sustainability models~

UPDATE: revised outline for Lesson 3 below

**Comment by s2t2** [comment]
Might be helpful / interesting to mention a list of open access journals, perhaps under section 2.3 > "Academic and research use cases". Like PLOS One https://journals.plos.org/plosone/ 

To motivate open publishing for a research audience, we can perhaps mention a use case / scenario of conducting a lit review and finding an awesome article only to not be able to access the article because it is behind a paywall.

**Comment by labarba** [comment]
We might have a separate tutorial focused on Open Access, but we might discuss here that researchers should look into the license used by the journal they are considered. 

Perhaps off-topic here (but on-topic for an Open Access tutorial), PLOS One follows the Gold OA model, with pricey [publishing fees](https://plos.org/publish/fees/): $2,290. (I'm not a fan of this model.) In contrast, journals of the Diamond OA model are free to publish and free to read. JOSS is a Diamond OA journal. (I'm a fan!)

**Comment by labarba** [comment]
# Jupyter book for this tutorial [WIP]

https://gw-ospo.github.io/oss-licensing/

**Comment by labarba** [comment]
**Lesson 1** on "Using Openly Licensed Works Published by Others" is done – 3865 words, ~0.5h reading time, estimate 1h live teaching time.

**Comment by labarba** [comment]
**Lesson 2** on "Licensing Your Own Software and Other Works" is done — 4168 words, ~0.5h reading time

**Comment by labarba** [comment]
Reviewing the first stab at an outline for Lesson 3 now... It needs to more tightly be in-scope with the overall topic of Open Source Licensing, so in regards to collaboration, we could edit out sub-topics like code sustainability, or coordination and communication overhead. Instead, let's bring in issues that may arise when many people contribute to a project: agreeing on the license early on, having a policy for contributions (e.g., will contributors be asked to license their work like the rest of the project?), working with people from different institutions, etc. We can also leave out the topic of community building.

**Comment by labarba** [comment]
**Lesson 3** started. First section and sub-sections as follows:

3.1 Establishing licensing policies for collaborative projects

- Agreeing on a license early in the project lifecycle
- Documenting and communicating the project's licensing policies
- Collaborations involving U.S. federal employees

**Comment by labarba** [comment]
3.2 Managing contributions and licensing in collaborative projects

- Ensuring contributions are properly licensed and compatible with the project's license
- Handling licensing issues when integrating third-party code or dependencies

UPDATE: adding 
- Automating license checks and compliance

**Comment by meganforbes** [comment]
Re: CLAs and DCOs - I know it's different from university to university, but possibly discussing how these might interact with an IP policy could be helpful? So, if the university owns copyright for anything produced with university resources, does the individual engineer have the right to sign a CLA or submit a DCO?
